### PR TITLE
Fixes #13953 - Can search Ostree Repos

### PR DESCRIPTION
**Pull request by parthaa**
Added an entry in the content menu to show ostree units

### CODE REVIEW

**Comment by parthaa** [comment]
The UI behavior changes for this PR documented in https://partha.fedorapeople.org/ostree-prs/PR-5894/

**Comment by daviddavis** [comment]
I tested and it works well. :+1: One small nitpick is that the commit SHA is wrapped when it doesn't need to be:

screen shot 2016-03-22 at 12 46 54 pm

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/details/ostree-branches-details-repositories.controller.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * @requires $scope
+ * @requires $q
+ * @requires Nutupane
+ * @requires Repository
+ * @requires Environment
+ * @requires ContentView
+ * @requires CurrentOrganization
+ *
+ * @description
+ *   Provides the functionality for the errata details repositories page.
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/details/ostree-branches-details-repositories.controller.js]: Need to update description.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/details/ostree-branches-details-repositories.controller.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ */
+angular.module('Bastion.ostree-branches').controller('OstreeBranchesDetailsRepositoriesController',
+    ['$scope', '$q', 'Nutupane', 'Repository', 'Environment', 'ContentView', 'CurrentOrganization',
+    function ($scope, $q, Nutupane, Repository, Environment, ContentView, CurrentOrganization) {
+        var repositoriesNutupane, environment, contentView;
+        var params = {
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/details/ostree-branches-details-repositories.controller.js]: All vars should be declared on one line.  I'm surprised this wasn't caught by our style checking @ehelms.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/details/ostree-branches-details.controller.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * @description
+ *   Provides the functionality for the Ostree Branch action pane.
+ */
+angular.module('Bastion.ostree-branches').controller('OstreeBranchesDetailsController', ['$scope', 'OstreeBranch',
+    function ($scope, OstreeBranch) {
+        if ($scope.branch) {
+            $scope.panel = {loading: false};
+        } else {
+            $scope.panel = {loading: true};
+        }
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/details/ostree-branches-details.controller.js]: There is a new pattern for this, see https://github.com/Katello/katello/blob/master/engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details.controller.js#L24-L31

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/ostree-branches.controller.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+/**
+ * @ngdoc object
+ * @name  Bastion.ostree-branches.controller:OstreeBranchesController
+ *
+ * @requires $scope
+ * @requires $location
+ * @requires translate
+ * @requires Nutupane
+ * @requires OstreeBranch
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/ostree-branches.controller.js]: Missing `Task`

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/packages/details/package-details-repositories.controller.js [hunk 5]
```diff
@@ -22,7 +22,7 @@ angular.module('Bastion.packages').controller('PackageDetailsRepositoriesControl
             'organization_id': CurrentOrganization
         };
 
-        repositoriesNutupane = new Nutupane(Repository, params, 'repositories');
+        repositoriesNutupane = new Nutupane(Repository, params);
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/packages/details/package-details-repositories.controller.js]: Is this change related and intentional?

File: engines/bastion_katello/test/ostree-branches/details/ostree-branches-details-repositories.controller.test.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        CurrentOrganization;
+
+    beforeEach(module('Bastion.ostree-branches', 'Bastion.test-mocks'));
+
+    beforeEach(inject(function($injector) {
+        var $controller = $injector.get('$controller'),
+            MockResource = $injector.get('MockResource'),
+            $q = $injector.get('$q');
+
+
```
Comment (waldenraines) [inline engines/bastion_katello/test/ostree-branches/details/ostree-branches-details-repositories.controller.test.js]: Nitpick: extra space.

File: engines/bastion_katello/test/ostree-branches/details/ostree-branches-details-repositories.controller.test.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            ContentView: ContentView,
+            Repository: Repository,
+            Environment: Environment,
+            CurrentOrganization: CurrentOrganization
+        });
+    }));
+
+    it("puts the ostree branches repositories table object on the scope", function() {
+        expect($scope.detailsTable).toBeDefined();
+    });
```
Comment (waldenraines) [inline engines/bastion_katello/test/ostree-branches/details/ostree-branches-details-repositories.controller.test.js]: This controller does a good deal more than you give it credit for in this test.  At the very least you need a test for `$scope.filterBranches`.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/details/ostree-branches-details.controller.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+        $scope.branch = OstreeBranch.get({id: $scope.$stateParams.branchId}, function () {
+            $scope.panel.loading = false;
+        });
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/details/ostree-branches-details.controller.js]: Meant to mention above that you need to follow the example [here ](https://github.com/Katello/katello/blob/master/engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details.controller.js#L33-L39) in order to remove the loading spinner on an error and display the error message.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/packages/details/package-details-repositories.controller.js (hunk 5 above)
Comment (parthaa) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/packages/details/package-details-repositories.controller.js]: @waldenraines this is intentional. I was trying to fix another bug with this change. If you go to Package Search and click on Repositories you get a js error . Look at 
Package Search

Comment (parthaa) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/packages/details/package-details-repositories.controller.js]: This change fixed ^ error. I noticed this when I based OSTree Branches search on Packages and had to fix in the other place.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/ostree-branches.controller.js (hunk 4 above)
Comment (parthaa) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/ostree-branches/ostree-branches.controller.js]: Task wasnt needed. Removed it .

File: engines/bastion_katello/test/ostree-branches/details/ostree-branches-details-repositories.controller.test.js (hunk 7 above)
Comment (parthaa) [inline engines/bastion_katello/test/ostree-branches/details/ostree-branches-details-repositories.controller.test.js]: Good point. Will add this

**Comment by ehelms** [comment]
@parthaa looks like you having failing tests here still

**Comment by parthaa** [comment]
@ehelms updated

**Comment by ehelms** [comment]
@parthaa I think you mean that for @waldenraines :)

**Comment by waldenraines** [comment]
ACK
